app/views.py

```diff
@@ -0,0 +1,339 @@
+
+def verify_password(email, password):
+    g.user = User.query.filter(email == User.email).first()
+    if g.user is not None and g.user.check_password(password):
+        return g.user
+    return False
+
+
+def identity(payload):
+    user_id = payload['identity']
+    return User.query.filter(user_id == User.user_id).first()
+
+
+jwt = JWT(app, verify_password, identity)
+
+
+def user_exists(email):
+    exists = db.session.query(db.session.query(User)
+                              .filter_by(email=email)
+                              .exists()).scalar()
+    return exists
+
+
+@app.route("/resource", methods=['POST'])
+@jwt_required()
+def get_resource():
+    return jsonify(dict(msg="Hello world"))
+
+
+def handle_error(error):
+    response = {}
+    status_code = 500
+    if isinstance(error, HTTPException):
+        status_code = error.code
+    response["status_code"] = status_code
+    response["error"] = str(error)
+    response['description'] = error.description
+    return jsonify(response), status_code
+
+
+# change error handler to return json instead of html
+for code in default_exceptions.keys():
+    app.errorhandler(code)(handle_error)
+
+
+@app.route("/auth/register", methods=['POST'])
+def register_user():
+    response = {}
+    json = request.json
+    validation = validate_register(json)
+    if validation.status:
+        if not user_exists(json["email"]):
+            user = User(
+                first_name=json["first_name"],
+                last_name=json["last_name"],
+                email=json["email"],
+                password=json["password"]
+            )
+            db.session.add(user)
+            db.session.commit()
+            status_code = 201
+            response["message"] = \
+                "Successfully registered the user with the email %s." % (json[
+                                                                         "email"])
+        else:
+            status_code = 409
+            response["message"] = \
+                "The user with the email %s already exists" % (json["email"])
+    else:
+        status_code = 400
+        response["message"] = validation.message
+
+    response = jsonify(response)
+    response.status_code = status_code
+    return response
+
+
+@app.route("/bucketlists", methods=['POST'])
+@jwt_required()
+def create_bucketlist():
+    '''Create a new bucket list'''
+    response = {}
+    json = request.json
+    validation = validate_bucketlist(json)
+    if validation.status:
+        bucketlist = BucketList(name=json['name'])
+        current_identity.bucketlists.append(bucketlist)
+        db.session.add(bucketlist)
+        db.session.commit()
+        status_code = 201
+    else:
+        status_code = 400
+
+    response["message"] = validation.message
+    response = jsonify(response)
+    response.status_code = status_code
+    return response
+
+
+@app.route("/bucketlists", methods=['GET'])
+@jwt_required()
+def get_all_bucketlists():
+    '''
```

Comment:
function `def get_all_bucketlists():` and  `def create_bucketlist():` could joined into one function with less repetitive code by `@app.route("/bucketlists", methods=['GET', 'POST'])`